sql/executor.go

```diff
@@ -190,6 +192,24 @@ func (e *Executor) execStmts(sql string, planMaker *planner) driver.Response {
 		// TODO(pmattis): Need to record the leases used by a transaction within
 		// the transaction state and restore it when the transaction is restored.
 		planMaker.releaseLeases(e.db)
+
+		// The previous statement finished executing a schema change. Wait for
+		// the schema change to propagate to all nodes, so that once the executor
+		// returns the new schema is live everywhere. This is not needed for
+		// correctness but is done to make the UI experience/tests predictable.
+		if planMaker.completedSchemaChange != nil {
+			for _, id := range planMaker.completedSchemaChange {
+				// To prevent Publish() from writing a new version of the
+				// table descriptor, the function passed in returns an error.
+				if err := planMaker.leaseMgr.Publish(id, func(desc *TableDescriptor) error {
```

Comment:
will that alloc some IDs? If we want this functionality, might be better to have something in the lease manager supporting this instead of this light abuse of `Publish()`.